mailbox/api/src/main/java/org/apache/james/mailbox/events/EventDeadLetters.java

```diff
@@ -27,7 +27,13 @@
 interface EventDeadLetters {
     Mono<Void> store(Group registeredGroup, Event failDeliveredEvent);
 
-    Flux<Event> take(Group registeredGroup);
+    Mono<Void> remove(Group registeredGroup, Event.EventId failDeliveredEventId);
 
-    Flux<Group> groups();
+    Mono<Event> failedEvent(Group registeredGroup, Event.EventId failDeliveredEventId);
+
+    Flux<Event> failedEvents(Group registeredGroup);
```

Comment:
Can we only return eventIds to match webadmin structure?

Reply:
I am afraid of we will find it helpful in the future when somewhere we do interacting with EventDeadletters in EventBus. and transform it into Flux<EventId> is easy.  I could write a default method.

Reply:
If we have a list of eventIds we can easily flatMap it into a list of events.

I'm affraid there is a significant overhead with loading the full Event JSON (parsing JSON) + read cost with payload being several KB.

We might be happy not to load all of that, all of the time.

Not to mention that the EventId proposal is closer to the WebAdmin API: that gonna be easier to implement!

Reply:
regarding having an optimal API for WebAdmin, I follow your idea